### ISSUE TITLE
Modular space for announcements and content

### ISSUE THREAD

**Issue by ajparsons** [mysociety/theyworkforyou#1689]
What I'd like is a more nuanced version of our banner announcements, where we can more easily roll out a rotation of things to promote through the site - and have these running all the time (the banner loses its impact if run all the time).

For instance, we might have multiple messages in rotation:

- Donate to support mySociety's work
- Read latest blog post [x]
- Take part in our user survey
- Cross-site promotion of CAPE/WDTK, etc

These messages would ideally be in the admin with a light-weight interface rather than deployed through the code. 

I'd like to integrate into:

- Homepage
- Email alerts
- MP summary pages

For a display pov, these different locations suggest slightly different designs. An extra complication is we might sometimes want some elements that have custom html or forms (e.g. subscribe to newsletter), we we might (see https://github.com/mysociety/supporters/issues/2) want to experiment with integrating donation forms directly into sites. This feels special-caseable once the basic setup is in place. 

In terms of the fields to store, an announcement has a:

- Title
- Short description
- URL
- Large image
- Thumbnail
- Format (for future special casing - e.g. 'link', 'donate', 'newsletter').
- Currently live (booelan)

The first steps is some light mock-ups of what this would look like, and then scope out with design what is possible with available time.

**Comment by zarino** [comment]
Notes from my discussion today with @lucascumsille:

**Potential locations on homepage:**

- Banner below the navbar (like Scorecards methodology banner)
- Replacing the (not used much) "random debate" and "topics in the news" below the hero

**Potential locations on MP page:**

- Banner below the navbar
- Replacing the yellow COVID panel
- Below the in-page nav sidebar (making the sidebar wider too)

@lucascumsille is going to mock up how a blog post (as just one example of a type of content) might look in those five positions, so we can see whether that sparks any ideas.

Meanwhile, we noted that donations feel like a different *kind* of content to blog posts etc – they feel less timely. @lucascumsille wondered whether we should treat them differently, visually. Thoughts, @ajparsons ?

We also wondered whether we actually *need* an image? Especially on our blog posts, the generic unsplash images always feel like filler. What’s the image for? If it’s to attract attention, we wondered whether that could that be achieved through colour alone?

**Comment by ajparsons** [comment]
I don't mind different approaches on the image - possibly a generic image for the blog series overall might be fine rather than for each post.

**Comment by lucascumsille** [comment]
I have made some mockups for the TWFY homepage:

### Banner for navbar
**Pros**

- We can announce any topics: Blogs, partner services, and donations.
- It could be present all the time, not only in certain places.
- Personally, I don't find it distracting from the main content. An easy way that the user can access that information after have scrolled the page/site.

### Cons

- Because of the space limit, an image wouldn't be a great idea. We could use an icon so the user can scan easily when it is a blog, service or donation; also, it would make it stand out a bit more without being too intrusive.

<img width="957" alt="Screenshot 2023-03-08 at 10 15 01" src="https://user-images.githubusercontent.com/13790153/223965952-cde95160-8727-4298-8a28-8ec6ab307d51.png">


### Using the current layout below the hero section

**Pros**

- Good to highlight the interesting content.

**Cons**

- I wouldn't use it for donations. For current users might not look bad, but for new users that might not even know what the page is about, asking for the donation in such a highlighted way might not sit well. Unless we use the Wikipedia technique when they are raising money and have a compelling web copy. More than a "Contribute to this project" and "donate" without new users not even having tried our service.

**Note:** I changed the layout for the component below the hero section. Personally not a huge fan of making content compete with each by locating them horizontally to each other, especially when there are a lot of interactive elements within them.

Here are a few mockups
<img width="1913" alt="Screenshot 2023-03-09 at 06 55 44" src="https://user-images.githubusercontent.com/13790153/223966147-c6be6b53-69f6-4056-8297-9d44784d38f6.png">
<img width="1913" alt="Screenshot 2023-03-09 at 07 00 33" src="https://user-images.githubusercontent.com/13790153/223966172-de8b1d83-5dcd-4b5a-91d3-3fefbdb279d0.png">
<img width="1913" alt="Screenshot 2023-03-09 at 07 38 31" src="https://user-images.githubusercontent.com/13790153/223966185-626cf3fd-efac-430b-a45d-f9f30cf8ca2b.png">
<img width="957" alt="Screenshot 2023-03-09 at 07 45 08" src="https://user-images.githubusercontent.com/13790153/223966189-432bb63b-909c-4822-8566-d2af35ba1960.png">

**Comment by ajparsons** [comment]
On the homepage, I'm confused on if we already did something like this in a previous wave and are just not using it, or if we didn't do that and just took things off the homepage at that time.

https://github.com/mysociety/theyworkforyou/issues/1532#issuecomment-793583285

**Comment by lucascumsille** [comment]
### MP page

**CTA main container**
Not sure how I feel about putting a CTA as first component for the content. Unless it was something really related to it.
<img width="1913" alt="Screenshot 2023-03-09 at 08 44 46" src="https://user-images.githubusercontent.com/13790153/223974038-1e637574-e7e1-461d-8f72-9d9aca80d13b.png">

**Side bar**
I prefer this approach. This would give us enough space to have both CTA (Donation + something else)
<img width="1913" alt="Screenshot 2023-03-09 at 08 55 46" src="https://user-images.githubusercontent.com/13790153/223974174-0b5eb321-163f-47da-8482-f07e6eba0970.png">
<img width="1913" alt="Screenshot 2023-03-09 at 08 49 25" src="https://user-images.githubusercontent.com/13790153/223974196-a0613400-96a0-478d-ba20-fbc909b44eb2.png">

**Comment by lucascumsille** [comment]
PD: With the first mock up I didn't change the yellow background colour, but it's unlikely to be yellow.

**Comment by ajparsons** [comment]
Sorry I missed this comment.

I think those sound like the correct approach (It's nice to have a few different choices for how to promote something to be sensitive to the length of the message).

**Comment by lucascumsille** [comment]
TODO for myself
- [ ] Having the top navbar banner, but for short messages, such as Donations or something that we really want to bring attention to. I could change the background colour to something more modest that pop-up less and still is present enough.
- [ ] Include the banner below the hero to display current and relevant info; that one line won't suffice.
- [ ] We can include the sidebar on the MP's pages.

@ajparsons 
Not sure if you know, but double checking do we already have a the first TODO implemented? I haven't checked, but I might need to tweak the style in case already exist.

**Comment by ajparsons** [comment]
So the current banner sits at the very top of the page:

https://github.com/mysociety/theyworkforyou/blob/598d660ede44a4c66b83ab9a25d50b4cfc6be0e0/www/includes/easyparliament/templates/html/header.php#L80

This could be moved and adapted. I can't see that we'd want *both*.

**Comment by ashapond** [comment]
I imagine this is already on your radar, but given the importance of the newsletter as a way for us to onboard new supporters, should 'sign up to our newsletter' be one of the messages on the navbar banner?
